usl_pipeline/cloud_functions/main.py

```diff
@@ -180,15 +180,17 @@ def build_and_upload_study_area_chunk(
             # File names should be in the form <study_area_name>/<file_name>
             study_area_name = file_name.parent.name
 
+            fd.seek(0)  # Seek to the beginning so the file can be read.
```

Comment:
it looks like we're copying the file unchanged between GCS locations, so another way to handle this would be to use [copy_blob](https://gcloud.readthedocs.io/en/latest/storage-buckets.html#google.cloud.storage.bucket.Bucket.copy_blob) rather than `upload_from_file(fd)`. It should also be cheaper.

Reply:
SG I'll work on replacing it with copy as an optimiziation. added a TODO